firedrake/functionspacedata.py

```diff
@@ -538,13 +540,28 @@ def get_map(self, V, entity_set, map_arity, bcs, name, offset, parent):
             else:
                 new_entity_node_list = entity_node_list
 
+            if kind == "interior_facet" and self.bt_masks is not None:
+                bt_masks = {}
+                off = map_arity // 2
+                for method, (bottom, top) in iteritems(self.bt_masks):
+                    b = []
+                    t = []
+                    for i in bottom:
+                        b.append(i)
+                        b.append(i+off)
+                    for i in top:
+                        t.append(i)
+                        t.append(i+off)
+                    bt_masks[method] = (b, t)
```

Comment:
the bt_masks argument to the `Map` constructor is a dict of pairs.  The first entry in the pair is the mask for the bottom, the second entry is the mask for the top.  Previously, the code assumed that for interior horizontal facets, the pair is the same as the cell pair.  Of course, this is wrong, because now we only mask out the dofs for the + cell.  This reconstructs the pair to apply the mask to everything on the bottom or top in the full map.